api/agent/call.go

```diff
@@ -205,11 +206,24 @@ func FromModel(mCall *models.Call) CallOpt {
 	return func(a *agent, c *call) error {
 		c.Call = mCall
 
-		// NOTE this adds content length based on payload length
 		req, err := http.NewRequest(c.Method, c.URL, strings.NewReader(c.Payload))
 		if err != nil {
 			return err
 		}
+
+		// Hoist original request content type into async request
+		if req.Header.Get("Content-Type") == "" {
+			content_type, ok := c.EnvVars["Fn_header_content_type"]
+			if ok {
+				req.Header.Set("Content-Type", content_type)
+			}
+		}
+
+		// Ensure content-length in async requests for protocol/http DumpRequestTo()
+		if req.Header.Get("Content-Length") == "" {
+			req.Header.Set("Content-Length", strconv.FormatInt(int64(len(c.Payload)), 10))
+		}
```

Comment:
no need to convert, just use `req.ContentLength`

Reply:
req.ContentLength does not work. How these headers are synthesized into actual headers depends on how/who processes this request. In this case, DumpRequestTo() in protocol handles this and fails since it only parses the actual Header map.